pkg/config/config.go

```diff
@@ -47,6 +47,36 @@ func (s *Secret) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return unmarshal((*plain)(s))
 }
 
+// NullBool will be used for boolean configuration values that should not
+// default to false when undefined and the default setting is true.
+type NullBool struct {
```

Comment:
Do we really need this? None of other boolean flags required this. Essentially if default is false, it's not needed, right? And we plan to use default false at the moment 🤔

Reply:
If we use a normal bool and the user sets the value to true in the defaults section, and leaves it blank in the receiver configs, the receiver config setting will actually be interpreted as false. By using NullBool, we are able to honor the global setting in the defaults section, which to me, seems more intuitive than allowing users to specify a default setting that does not actually change the behavior of the application.

Reply:
Got it, thanks--good point about that logic. However wouldn't using *bool fix the problem. Also if we fix it for this field let's do this for all bools that can be in default fields for consistency. 

Alternatively we could make it a bool (and accept the bug) and move those bools to *bool or your struct in separate PR, wdyt? I would love to avoid complexity of some fields of the same type having different logic.

Reply:
I updated the PR to use bool* instead of NullBool and applied the same logic to AddGroupLabels so that default will be honored.